snapcraft/internal/lxd/_containerbuild.py

```diff
@@ -195,6 +196,8 @@ def _container_run(self, cmd, cwd=None, **kwargs):
         if sh:
             cmd = ['sh', '-c', '{}{}'.format(sh,
                    ' '.join(pipes.quote(arg) for arg in cmd))]
+        if user:
+            cmd = ['sudo', '-H', '-E', '-u', self._user] + cmd
```

Comment:
I'm afraid I still find this confusing. We use the phrase "set up a user" and "run as user" and now we have "user is true or false" and yet by default self._user is 'root'. :stuck_out_tongue:

Reply:
I changed it to a `user='foo'` argument. I'm actually liking this best now.